jax/interpreters/xla.py

```diff
@@ -59,6 +59,32 @@
 XlaComputationBuilder = Any  # xla_bridge._JaxComputationBuilder
 XlaExecutable = Any # xla_extension.LocalExecutable
 
+
+class NameStackFrame(NamedTuple):
```

Comment:
I think it is worth considering separating this mechanism from the generation of XlaOpMetadata. The latter can be based on this, but maybe not quite so literally. For example, we may prefer `jit(foo)` rather than `xla_call(foo)`. For `named_call` primitive we may prefer to just say `foo` rather than `named_call(foo)`. In essence, we should not require that we surface the internal names of the primitives in the XlaOpMetadata.

Reply:
Ok, but how would you suggest we do that? Implement an XLA specific pretty-printer of those objects instead of relying on their `__str__` methods?

Reply:
The cleanest way would be indeed to implement a NameStack pretty-printer for XLA. Or, if the only customization is the name of the primitive, we can go half-way there by passing an optional "op_metadata_name" field.  If the field is empty then we drop the "()" also, so that we can print `foo` instead of `named_call(foo)`.